pkg/controller/operators/openshift/helpers.go

```diff
@@ -168,24 +170,39 @@ func incompatibleOperators(ctx context.Context, cli client.Client) (skews, error
 	return incompatible, nil
 }
 
-func desiredRelease(ctx context.Context, cli client.Client) (*semver.Version, error) {
-	cv := configv1.ClusterVersion{}
-	if err := cli.Get(ctx, client.ObjectKey{Name: "version"}, &cv); err != nil { // "version" is the name of OpenShift's ClusterVersion singleton
-		return nil, &transientError{fmt.Errorf("failed to get ClusterVersion: %w", err)}
+var (
+	openshiftRelease *semver.Version
+	releaseMutex     sync.Mutex
```

Comment:
I think I have some reservations around a package-level mutex. What happens if multiple places are locking and unlocking this mutex?